pebble/pathapp/src/pathapp.c

```diff
@@ -0,0 +1,285 @@
+#include "pebble.h"
+
+#define MAXBUFFERSIZE 200
+#define MAX_SCHED_ITEMS 10
+#define MAX_STN_ITEMS 20
+#define MAX_DIR_ITEMS 5
+
+#define STN_NAME_LEN 10
+
+static Window *main_window, *direction_window, *schedule_window;
+static SimpleMenuLayer *stn_layer, *dir_layer, *sched_layer;
+
+static SimpleMenuItem stn_items[MAX_STN_ITEMS], dir_items[MAX_DIR_ITEMS], sched_items[MAX_SCHED_ITEMS];
+static SimpleMenuSection stn_sections[1], dir_sections[1], sched_sections[1];
+
+static char buffer[MAXBUFFERSIZE];
+
+static char from_station[STN_NAME_LEN];
+static char *delim = ";";
+
+enum {
+	PATH_STN_KEY = 0x0,
+};
+
+static size_t sched_num_items;
+
+static char *stn_names[] = {
+	"JSQ",
+	"Grove St",
+	"Newport",
+	"Hoboken",
+	"Chris St",
+	"9 St",
+	"14 St",
+	"23 St",
+	"33 St",
+	NULL,
+};
+
+struct direction {
+	char *title;
+	char *stub;
+};
+
+static struct direction directions[] = {
+	{ .title = "JSQ to 33rd", .stub = "jsq_33rd" },
+	{ .title = "33rd to JSQ", .stub = "33rd_jsq" },
+};
+
+static void send_to_phone(int key, const char *direction) {
+	DictionaryIterator *iter;
+	char *buf;
+
+	app_message_outbox_begin(&iter);
+	if (!iter) {
+		return;
+	}
+
+	buf = buffer;
+	memcpy(buf, from_station, strlen(from_station));
+	buf += strlen(from_station);
+	memcpy(buf, delim, strlen(delim));
+	buf += strlen(delim);
+	strncpy(buf, direction, strlen(direction));
+
+	buf = buffer;
+	Tuplet tuplet = TupletCString(key, buf);
+	dict_write_tuplet(iter, &tuplet);
+
+	dict_write_end(iter);
+
+	app_message_outbox_send();
+
+	// Zero out the buffer area we used. We can leave the last
+	// NUL byte as it is, so we don't need to do strlen(buffer)+1 here.
+	memset(buffer, 0x00, strlen(buffer));
+}
+
+static void sched_menu_callback(int index, void *ctx) {
+	vibes_short_pulse();
+	sched_items[index].subtitle = "Ok. Now hurry!";
```

Comment:
Can you declare all the app messages as static strings somewhere or even an enum for all messages is fine.